packages/admin/admin/src/dateTime/DateTimeRangePicker.tsx

```diff
@@ -0,0 +1,175 @@
+import { type DateTimeRangePickerProps as MuiDateTimeRangePickerProps } from "@mui/x-date-pickers-pro";
+import { type ComponentType, lazy, type ReactNode, Suspense, useState } from "react";
+
+import { ClearInputAdornment as CometClearInputAdornment } from "../common/ClearInputAdornment";
+import { OpenPickerAdornment } from "../common/OpenPickerAdornment";
+import { ReadOnlyAdornment } from "../common/ReadOnlyAdornment";
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { type ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+import { isValidDate } from "./utils";
+
+export type DateTimeRange = {
+    start: Date | null;
+    end: Date | null;
+};
+
+export type DateTimeRangePickerClassKey = "root" | "clearInputAdornment" | "readOnlyAdornment" | "openPickerAdornment";
+
+export type DateTimeRangePickerProps = ThemedComponentBaseProps<{
+    root: ComponentType<MuiDateTimeRangePickerProps<Date, true>>;
+    clearInputAdornment: typeof CometClearInputAdornment;
+    readOnlyAdornment: typeof ReadOnlyAdornment;
+    openPickerAdornment: typeof OpenPickerAdornment;
+}> & {
+    fullWidth?: boolean;
+    required?: boolean;
+    value?: DateTimeRange;
+    onChange?: (date: DateTimeRange | undefined) => void;
+    iconMapping?: {
+        openPicker?: ReactNode;
+    };
+} & Omit<MuiDateTimeRangePickerProps<Date, true>, "value" | "onChange">;
+
+export const DateTimeRangePicker = (inProps: DateTimeRangePickerProps) => {
+    const {
+        iconMapping = {},
+        fullWidth,
+        required,
+        slotProps,
+        disabled,
+        value: valueObject,
+        onChange,
+        readOnly,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminDateTimeRangePicker",
+    });
+    const [open, setOpen] = useState(false);
+    const arrayValue: [Date | null, Date | null] = valueObject ? [valueObject.start, valueObject.end] : [null, null];
+
+    const { openPicker: openPickerIcon = <Calendar color="inherit" /> } = iconMapping;
+
+    return (
+        <Suspense>
+            <LazyRoot
+                enableAccessibleFieldDOMStructure
+                disabled={disabled}
+                readOnly={readOnly}
+                open={open}
+                onOpen={() => setOpen(true)}
+                onClose={() => setOpen(false)}
+                disableOpenPicker
+                value={arrayValue}
+                onChange={([startDate, endDate]) => {
+                    const startDateIsInvalid = startDate !== null && !isValidDate(startDate);
+                    const endDateIsInvalid = endDate !== null && !isValidDate(endDate);
+
+                    if (startDateIsInvalid || endDateIsInvalid) {
+                        return;
+                    }
+
+                    if (!startDate && !endDate) {
+                        onChange?.(undefined);
+                        return;
+                    }
```

Comment:
The DateTimeRange supports partial ranges (e.g., no end), should we support it here as well?

Reply:
We do support it, only when both values are `null`, the value is set to `undefined` instead of the object.